fili-core/src/main/java/com/yahoo/bard/webservice/druid/serializers/Util.java

```diff
@@ -44,4 +45,25 @@
         // If we cannot find the physical name, then return empty optional.
         return Optional.empty();
     }
+
+    /**
+     * JSON tree walk to determine if there is a nested query below the current json node or not.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on
+     *
+     * @return  a Boolean where true indicates there are more nested query below this node, false otherwise
+     */
+    public static Boolean moreNestedQuery(JsonGenerator gen) {
+        JsonStreamContext context = gen.getOutputContext();
+
+        while (context != null) {
+            Object parent = context.getCurrentValue();
+            if (parent instanceof DruidQuery) {
```

Comment:
Can we extract the "tree walk to find the next level up" into it's own method? Both of the helper methods in this class are currently doing that work. `Optional<DruidQuery>` may be a useful return type to use, since it may let us keep things clean.

Here's one (untested) idea:
```java
public <T> Optional<T> withContainingQuery(JsonGenerator gen, Function<DruidQuery, T> success) {
    JsonStreamContext context = gen.getOutputContext();
    while (context != null) {
        Object parent = context.getCurrentValue();
        if (parent instanceof DruidQuery) {
            return Optional.of(success.apply((DruidQuery) parent));
        }
        context = context.getParent();
    }
    return Optional.empty();
}
```